### DISCUSSION TITLE
Open editor only for text files

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [junegunn/fzf#3698] [Q&A] [answered]
How can I use fzf to open an editor only if the selected file is a text file? If it's not a text file, I want it to go back to fzf for another selection. Additionally, once the file is opened in the editor, I want the script to exit when the editor is closed.

**Comment by LangLangBart** [comment, marked as answer]
One idea is to use the `transform`[^1] action to check if the file meets certain requirements. The
check below has been used by the maintainer in one of their scripts[^2] to verify if it's a text
file.

```bash
[[ -f {} && ! $(file --dereference --mime -- {}) =~ binary ]]
```

If the file meets the requirements, perform the desired action, such as `become:vim …`[^3], or do
nothing.

> [!TIP]
> The placeholder, for instance `{}`, needs to be escaped in the `echo` statement for `transform`
> actions. Otherwise it will cause issues with files that contain spaces in their names.
>
>
> ✅ `echo 'become:vim \{}'`
>
> ❌ `echo 'become:vim {}'`

```bash
fzf --bind "enter:transform:[[ -f {} && ! \$(file --dereference --mime -- {}) =~ binary ]] &&
  echo 'become:vim \{}' ||
  echo 'change-header:I am not a text file'"
```

<img src="https://github.com/junegunn/fzf/assets/92653266/2d8523d7-e5c3-4413-a546-a42629b34917" width="800">

[^1]: [CHANGELOG.md 0.45.0](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0450)
[^2]: [fzf/bin/fzf-preview.sh](https://github.com/junegunn/fzf/blob/master/bin/fzf-preview.sh#L17-L33)
[^3]: [CHANGELOG.md 0.38.0](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0380)

**Reply by limaceous-bushwhacker** [reply]
This is amazing, ty! I don't fully understand what's going on here since I'm not familiar with `transform` and `become` syntax but thanks for introducing me to these.
